core/src/main/java/com/linecorp/armeria/common/LeakDetectionConfiguration.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.common.util.Sampler;
+
+/**
+ * The configuration that determines whether to trace request context leaks and how frequently to
+ * keeps stack trace.
+ */
+public final class LeakDetectionConfiguration {
```

Comment:
Can we just replace this method with `Sampler<? super RequestContext>`? We could check the sampler `== Sampler.never()` to see if it's disabled. If we're concerned about a user using his/her own `never` implementation, we might want to add `Sampler.isNever()`.